### PR TITLE
[TZP] remove moment-timezone dependency, move into datetime

### LINKED ISSUE #2120: [timezone] remove moment-timezone dependency?
here's what I'd like to do with `TimezonePicker` for 2.0:

1. remove `moment-timezone` dependency
1. add `timezones: ITimezoneItem[]` required prop: user passes array of known timezones from whatever library they wish
    - this should replace `private timezoneItems` and `private initialTimezoneItems` 
    - maybe add a prop to `ITimezoneItem` to indicate if it should appear in initial items? (item shown when query is empty)
1. move component back into `datetime` pkg since it no longer has this extra huge dependency
1. kill `timezone` package

### PR DESCRIPTION
**Pull request by giladgray**
#### Checklist

- [ ] Include tests
- [ ] Update documentation

#### Changes proposed in this pull request:

- move TZP into datetime package
- add required `timezones` prop for list of all timezones
- remove uncontrolled props - only controlled usage

#### Reviewers should focus on:

- some props still need to be removed/worked around
- filtering currently depends on fuzz-aldrin-plus so we probably still need to remove that dependency

cc @brieb 👋

### CODE REVIEW

File: packages/datetime/src/timezone/timezonePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    /**
+     * Timezone list displayed when the query is empty.
+     * If this prop is omitted, the first `maxResults` entries in `timezones` will be shown.
+     */
+    initialTimezones?: ITimezoneItem[];
+
+    /**
+     * The currently selected timezone, e.g. "Pacific/Honolulu".
+     * If this prop is provided, the component acts in a controlled manner.
+     * https://en.wikipedia.org/wiki/Tz_database#Names_of_time_zones
```
Comment (adidahiya) [inline packages/datetime/src/timezone/timezonePicker.tsx]: can we reference a more authoritative list like https://www.iana.org/time-zones

File: packages/docs-app/src/examples/timezone-examples/momentTimezones.ts [hunk 2]
```diff
@@ -84,41 +68,43 @@ function getPopulousTimezoneItems(date: Date): ITimezoneItem[] {
...
-                return timezone1 < timezone2 ? -1 : 1;
-            }
-            // Descending order of population
-            return population2 - population1;
-        }
-        // Ascending order of offset
-        return offset1 - offset2;
+        // Order by offset ascending, population descending, timezone name ascending.
+        return offset1 === offset2
+            ? population1 === population2 ? (timezone1 < timezone2 ? -1 : 1) : population2 - population1
```
Comment (adidahiya) [inline packages/docs-app/src/examples/timezone-examples/momentTimezones.ts]: not a fan of nested ternarys. also you have lost useful comments

File: packages/datetime/src/timezone/timezoneItem.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    /** Display name for this timezone. Defaults to `timezone` property. */
+    displayName?: string;
+
+    /** Optional icon for the timezone. */
+    iconName?: IconName;
+
+    /** React key to use for the rendered element. Defaults to `timezone` property. */
+    key?: string;
+
+    /** Label for the timezone. */
```
Comment (adidahiya) [inline packages/datetime/src/timezone/timezoneItem.ts]: what would a blueprint consumer use this for?

File: packages/datetime/src/timezone/timezoneItem.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+     *
+     * This property is used as the identifier for a timezone item, so it should be unique.
+     * If you can't guarantee uniqueness, the `key` property can override it.
+     */
+    timezone: string;
+
+    /** Display name for this timezone. Defaults to `timezone` property. */
+    displayName?: string;
+
+    /** Optional icon for the timezone. */
```
Comment (adidahiya) [inline packages/datetime/src/timezone/timezoneItem.ts]: what's an example use case for this?

File: packages/datetime/src/timezone/timezoneItem.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+/** Timezone-specific QueryList item */
+export interface ITimezoneItem {
+    /**
+     * The actual timezone name, like `Europe/Paris`.
+     *
+     * This property is used as the identifier for a timezone item, so it should be unique.
+     * If you can't guarantee uniqueness, the `key` property can override it.
+     */
+    timezone: string;
```
Comment (adidahiya) [inline packages/datetime/src/timezone/timezoneItem.ts]: I would probably just call this `name: string`

File: packages/datetime/src/timezone/timezoneItem.ts (hunk 4 above)
Comment (giladgray) [reply packages/datetime/src/timezone/timezoneItem.ts]: `locate` for current timezone

File: packages/datetime/src/timezone/timezoneItem.ts (hunk 3 above)
Comment (giladgray) [reply packages/datetime/src/timezone/timezoneItem.ts]: anything on the right side. example uses it to show offset "-12:00"

Comment (giladgray) [reply packages/datetime/src/timezone/timezoneItem.ts]: needs better doc

Comment (adidahiya) [reply packages/datetime/src/timezone/timezoneItem.ts]: ok great, can you document that?

File: packages/datetime/src/timezone/timezoneItem.ts (hunk 5 above)
Comment (giladgray) [reply packages/datetime/src/timezone/timezoneItem.ts]: this component is getting dangerously close to nothing more than a `Select` with some default props...

File: packages/datetime/src/timezone/timezoneItem.ts (hunk 4 above)
Comment (adidahiya) [reply packages/datetime/src/timezone/timezoneItem.ts]: can we constrain the API here? I don't think it's necessary to allow any arbitrary icon. just ask the user to supply the current timezone (or detect it by default using native APIs) and show the `locate` icon yourself

Comment (giladgray) [reply packages/datetime/src/timezone/timezoneItem.ts]: yeah considered keeping around the `showCurrentTimezone` option. will see if i can bring it back. i removed cuz it was awkward to force an item in there that might look totally different from user-supplied items (like, it's in english but they gave me french items).

File: packages/datetime/src/timezone/timezoneItem.ts (hunk 5 above)
Comment (giladgray) [reply packages/datetime/src/timezone/timezoneItem.ts]: i might prefer `value` to match the controlled prop name?

Comment (adidahiya) [reply packages/datetime/src/timezone/timezoneItem.ts]: I guess it adds fuzzaldrin filtering on top too?

Comment (giladgray) [reply packages/datetime/src/timezone/timezoneItem.ts]: but do we want to include that dependency in `datetime`? i just removed it locally...

**Comment by giladgray** [comment]
🙅‍♂️ closing in favor of `timezone` package backed by `spacetime` library

**Comment by brieb** [comment]
👋 @giladgray 
Thanks for helping graduate the timezone component!
Nice to know about `spacetime` - will keep that in mind. 🙂

**Pull request closed without merging**